components/atomic/buttons/ButtonControl/patterns/ButtonControlConfirm.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { forwardRef } from "react";
+import { ButtonControl } from "components/atomic";
+import { useDialogState, DialogDisclosure } from "reakit/Dialog";
+import Modal from "components/layout/Modal";
+import ConfirmModalBody from "components/composed/ConfirmModalBody";
+
+type BaseProps = React.ComponentProps<typeof ButtonControl>;
+type ModalProps = React.ComponentProps<typeof Modal>;
+
+/**
+ * A button control that opens a modal
+ */
+const ButtonControlConfirm = forwardRef(
+  (
+    {
+      modalLabel,
+      modalBody,
+      icon,
+      children,
+      onClick,
+      "aria-label": actionLabel,
+      onOpenModal: closeDropdown,
+    }: Props,
+    ref
+  ) => {
+    const dialog = useDialogState({ visible: false, animated: true });
+
+    return (
+      <>
+        <ButtonControl
+          as="a"
+          icon={icon}
+          ref={ref}
+          onClick={(e) => {
```

Comment:
This took me too long to figure out, but I learned that you can't use Reakit's `<DialogueDisclosure>` and a click handler on the same element, so I had to create my own handler here to both close the dropdown and open the modal. I think the aria properties are all preserved, but I'd love your eye on this @jen-castiron!